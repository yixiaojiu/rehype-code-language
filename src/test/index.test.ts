import { test, expect } from 'vitest'
import { unified } from 'unified'
import remarkParse from 'remark-parse'
import remarkRehype from 'remark-rehype'
import rehypeStringify from 'rehype-stringify'
import rehypeCodeLanguage from '../index'

test('render span element with the content of js', async () => {
  const md = `
  # Some title
  
  \`\`\`js
  let a = 1;

  console.log(a);
  \`\`\`
  
  Some block of text
  `

  const result = await unified()
    .use(remarkParse)
    .use(remarkRehype)
    .use(rehypeStringify)
    .use(rehypeCodeLanguage)
    .process(md)

  expect(result.value).toContain('<span class="language">js</span>')
  expect(result.value).matchSnapshot()
})

test("don't render span element when there is no language in code block", async () => {
  const md = `
  # Some title
  
  \`\`\`
  some text
  \`\`\`
  
  Some block of text
  `

  const result = await unified()
    .use(remarkParse)
    .use(remarkRehype)
    .use(rehypeStringify)
    .use(rehypeCodeLanguage)
    .process(md)

  expect(result.value).not.toContain('<span class="language"></span>')
})

test('render span element with default content', async () => {
  const md = `
  # Some title
  
  \`\`\`
  some text
  \`\`\`
  
  Some block of text
  `

  const result = await unified()
    .use(remarkParse)
    .use(remarkRehype)
    .use(rehypeStringify)
    .use(rehypeCodeLanguage, { defaultLanguage: 'txt' })
    .process(md)

  expect(result.value).toContain('<span class="language">txt</span>')
  expect(result.value).matchSnapshot()
})
