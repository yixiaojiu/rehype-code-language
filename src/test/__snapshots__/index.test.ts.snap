// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render span element with default content 1`] = `
"<h1>Some title</h1>
<pre><span class="language">txt</span><code>some text
</code></pre>
<p>Some block of text</p>"
`;

exports[`render span element with the content of js 1`] = `
"<h1>Some title</h1>
<pre><span class="language">js</span><code class="language-js">let a = 1;

console.log(a);
</code></pre>
<p>Some block of text</p>"
`;
